### PR TITLE
Fix/25510

### LINKED ISSUE #25510: Checkout Address_2 field has duplicate 'optional'
**Describe the bug**
The label for address_2 has a duplication of '(optional)'. 
It's in the field label, then also added in a `<span>` tag.

**To Reproduce**
1. Goto WooCommerce.com
2. Goto checkout.
3. Disable the css on the `.screen-reader-text` CSS class - or view source to see the markup.

**Screenshots**
Screenshot 2020-01-28 at 17 31 03

**Expected behavior**
Just one '(optional)'

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by itzmekhokan**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes #25510  .

### Other information:

* [x] Have you successfully run tests with your changes locally?

### CODE REVIEW

File: includes/class-wc-checkout.php [hunk 1]
```diff
@@ -271,7 +271,7 @@ public function get_checkout_fields( $fieldset = '' ) {
 			// Add accessibility labels to fields that have placeholders.
 			foreach ( $fields as $single_field_type => $field ) {
 				if ( empty( $field['label'] ) && ! empty( $field['placeholder'] ) ) {
-					$this->fields[ $field_type ][ $single_field_type ]['label']       = $field['placeholder'];
+					$this->fields[ $field_type ][ $single_field_type ]['label'] = ( 'billing_address_2' === $single_field_type && 'optional' === get_option( 'woocommerce_checkout_address_2_field', 'optional' ) ) ? __( 'Apartment, suite, unit etc.', 'woocommerce' ) : $field['placeholder'];
```
Comment (ObliviousHarmony) [inline includes/class-wc-checkout.php]: I think it would be better to change this string at its source in: https://github.com/woocommerce/woocommerce/blob/d3a33f8d0547b709b4c7d7753b209802e85c3f44/includes/class-wc-countries.php#L650

Since the "optional" span isn't added unless the field isn't required, that should probably be all that needs done?

File: assets/js/admin/meta-boxes-order.js [hunk 2]
```diff
@@ -549,45 +549,31 @@ jQuery( function ( $ ) {
 		},
 
 		add_fee: function() {
-			window.wcTracks.recordEvent( 'order_edit_add_fee_click', {
```
Comment (ObliviousHarmony) [inline assets/js/admin/meta-boxes-order.js]: Please don't remove tracking events!

File: assets/js/admin/meta-boxes-order.js [hunk 3]
```diff
@@ -549,45 +549,31 @@ jQuery( function ( $ ) {
 		},
 
 		add_fee: function() {
```
Comment (ObliviousHarmony) [inline assets/js/admin/meta-boxes-order.js]: I would say that the original behavior here is okay, and instead the focus should be on resolving the server-side parsing in: https://github.com/woocommerce/woocommerce/blob/1d1413b4ee669cd37ee025c93972ce785a10a0b9/includes/class-wc-ajax.php#L995

It's worth noting that there does seem to be inconsistencies across WC in whether we respect the currency symbol or not. In this case, we should probably just remove the default currency symbol from the string before converting it to a float value.

File: includes/class-wc-ajax.php [hunk 4]
```diff
@@ -978,33 +978,8 @@ public static function add_order_fee() {
 				throw new Exception( __( 'Invalid order', 'woocommerce' ) );
 			}
 
-			$amount = isset( $_POST['amount'] ) ? wc_clean( wp_unslash( $_POST['amount'] ) ) : 0;
```
Comment (ObliviousHarmony) [inline includes/class-wc-ajax.php]: Sorry, could you give a little context around the decision to remove all of this?

**Review by ObliviousHarmony — CHANGES_REQUESTED** [review]
Thank you very much for your contribution! I've left some comments on the PR for you to review.

I noticed that you tried to sneak two fixes in here! Could you please resolve each issue in their own pull request?

**Comment by claudiosanches** [comment]
Sorry, but seems like you mixed some fixes on this PR.
I'll close it for now, please let me know if you want to write a new PR targeting only fixes for #25510

**Comment by stuart-lambon** [comment]
Hi, I wanted to have a go at solving this for my first issue but I can see it’s slightly more complicated that first thought.

The (optional) text should be added here:

https://github.com/woocommerce/woocommerce/blob/86311a9903eae582a3367acf48628bd0c775548d/includes/wc-template-functions.php#L2646

Rather than in the text here:

https://github.com/woocommerce/woocommerce/blob/86311a9903eae582a3367acf48628bd0c775548d/includes/class-wc-countries.php#L653

This works to remove the double `(optional)` issue. However, you cannot really do this otherwise it will remove the `(optional)` from the field placeholder too. The main problem is that `address_2` does not have a label so it uses the placeholder instead.

Should I remove the (optional) from the location that the placeholder is converted to a label or is there a better way?

https://github.com/woocommerce/woocommerce/blob/86311a9903eae582a3367acf48628bd0c775548d/includes/class-wc-checkout.php#L274

I can see someone else tried this route but it seems a little hacky to me:
https://github.com/woocommerce/woocommerce/pull/25621
